org.uqbar.project.wollok.typeSystem/src/org/uqbar/project/wollok/typesystem/UnionType.xtend

```diff
@@ -5,9 +5,9 @@ import org.eclipse.xtend.lib.annotations.Accessors
 
 class UnionType extends BasicType {
 	@Accessors(PUBLIC_GETTER)
-	List<AbstractContainerWollokType> types
+	List<ConcreteType> types
```

Comment:
AbstractContainerWollokType is an implementation helper and should not be used for dispatch, ConcreteType is the conceptually correct replacement.